web/lib/supabase/contracts.ts

```diff
@@ -117,17 +119,106 @@ export async function getContractVisibilityFromSlug(contractSlug: string) {
   return undefined
 }
 
-export async function searchContract(query: string) {
-  const { data } = await run(
-    db
-      .from('contracts')
-      .select('data')
-      .textSearch('question', `${query}`)
-      .limit(10)
-  )
+export async function searchContract(props: {
+  state: stateType
+  query: string
+  filter: filter
+  sort: Sort
+  offset: number
+  limit: number
+  group_id?: string
+  creator_id?: string
+}) {
+  const { state, query, filter, sort, offset, limit, group_id, creator_id } =
+    props
+
+  if (!query || query.length == 0) {
```

Comment:
Technically `''` is falsey, so the first condition is enough. But also, query has to be a string (based on the type), so the second condition would also be sufficient.